modular_citadel/code/modules/arousal/organs/penis.dm

```diff
@@ -18,23 +18,56 @@
 	var/knot_girth_ratio 	= KNOT_GIRTH_RATIO_DEF
 	var/list/dickflags 		= list()
 	var/list/knotted_types 	= list("knotted", "barbed, knotted")
+	var/prev_size			= 6 //really should be renamed to prev_length
+
+/obj/item/organ/genital/penis/Initialize()
+	. = ..()
+	/* I hate genitals.*/
 
 /obj/item/organ/genital/penis/update_size()
-	if(length == cached_length)
+	var/mob/living/carbon/human/o = owner
+	if(!ishuman(o) || !o)
 		return
-	switch(length)
-		if(-INFINITY to 5)
+	if(cached_length < 0)//I don't actually know what round() does to negative numbers, so to be safe!!
+		var/obj/item/organ/genital/penis/P = o.getorganslot("penis")
+		to_chat(o, "<span class='warning'>You feel your tallywacker shrinking away from your body as your groin flattens out!</b></span>")
+		P.Remove(o)
+	switch(round(cached_length))
+		if(0 to 4) //If modest size
+			length = cached_length
 			size = 1
-		if(5 to 9)
+			if(owner.has_status_effect(/datum/status_effect/chem/PElarger))
```

Comment:
These could be processed On_mob_life for this chemical. Just increment with P.size ++ and call to update the appearance. 

for PShrinker do P.size --

Reply:
Is this not wanted here in case someone else wants to do size changes too?